contrib/googlejavaformat/src/python/pants/contrib/googlejavaformat/googlejavaformat.py

```diff
@@ -0,0 +1,76 @@
+# coding=utf-8
+# Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+from abc import abstractproperty
+
+from pants.backend.jvm.tasks.scala_rewrite_base import ScalaRewriteBase
+from pants.base.exceptions import TaskError
+from pants.java.jar.jar_dependency import JarDependency
+from pants.task.fmt_task_mixin import FmtTaskMixin
+from pants.task.lint_task_mixin import LintTaskMixin
+
+
+class GoogleJavaFormatBase(ScalaRewriteBase):
+
+  _MAIN = 'com.google.googlejavaformat.java.Main'
+  _SCALA_SOURCE_EXTENSION = '.java'
+  _TARGET_TYPES = ['java_library', 'junit_tests', 'java_tests']
+
+  @classmethod
+  def register_options(cls, register):
+    super(GoogleJavaFormatBase, cls).register_options(register)
+    cls.register_jvm_tool(register,
+                          'google-java-format',
+                          classpath=[
+                          JarDependency(org='com.google.googlejavaformat',
+                                        name='google-java-format',
+                                        rev='1.5')
+                          ])
+
+  @classmethod
+  def implementation_version(cls):
+    return super(GoogleJavaFormatBase, cls).implementation_version() + [('GoogleJavaFormatBase', 1)]
+
+  def invoke_tool(self, _, target_sources):
+    args = list(self.additional_args)
+    args.extend([source for _, source in target_sources])
+    return self.runjava(classpath=self.tool_classpath('google-java-format'),
+                        main=self._MAIN,
+                        args=args,
+                        workunit_name='google-java-format',
+                        jvm_options=self.get_options().jvm_options)
+
+  @abstractproperty
+  def additional_args(self):
+    return []
```

Comment:
A bit contradictory to be @abstractproperty and have an implementation. How about hanging doc off the method instead? Alternatively, just `pass`.